packages/common/src/hooks/useFeatureFlag.ts

```diff
@@ -111,3 +115,57 @@ export const createUseFeatureFlagHook =
       setOverride
     }
   }
+
+/** Fetches enabled status of a given feature flag with fallback. Result is memoized. */
+export const useFeatureFlag = (
+  flag: FeatureFlags,
+  fallbackFlag?: FeatureFlags
+) => {
+  const overrideKey = `${FEATURE_FLAG_OVERRIDE_KEY}:${flag}`
+  const configLoaded = useHasConfigLoaded()
+  const { localStorage, remoteConfig } = useAppContext()
+
+  const shouldRecompute = useRecomputeToggle(
+    useHasAccount,
+    configLoaded,
+    remoteConfig
+  )
+
+  const isEnabled = useMemo(
+    () => remoteConfig.getFeatureEnabled(flag, fallbackFlag),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [flag, fallbackFlag, shouldRecompute]
+  )
+
+  const setOverride = useCallback(
+    async (value: OverrideSetting) => {
+      return value === null
```

Comment:
Worth noting: In the `createUseFeatureFlagHook` version above, we were accepting implementations of `getLocalStorageItem` and `setLocalStorageItem` via arguments.[ The web version of `useFlag`](https://github.com/AudiusProject/audius-protocol/blob/8ffcd2a070ae4f2a9b168fc82d991e75932a90c6/packages/web/src/hooks/useRemoteConfig.ts#L16) was passing a wrapped version of `setItem` that would check for null and call `removeItem` instead.[ The mobile version](https://github.com/AudiusProject/audius-protocol/blob/8ffcd2a070ae4f2a9b168fc82d991e75932a90c6/packages/mobile/src/hooks/useRemoteConfig.ts#L18) just passed `AsyncStorage.setItem` directly, which does _not_ check the value and automatically remove it.

Since we're now using the unified `LocalStorage` class here, which exposes the standard localStorage interface, I've moved that automatic removal logic here, so that our localStorage adapter maintains the standard interface.

Reply:
just fyi @nicoback @nicoback2